src/algorithms/distancegradient.jl

```diff
@@ -0,0 +1,25 @@
+"""
+    DistanceGradient
+
+The `sources` field is the *linear* indices of the matrix, from which the
+distance must be calculated.
+"""
+struct DistanceGradient <: NeutralLandscapeMaker
+    sources::Vector{Integer}
+end
+
+DistanceGradient() = DistanceGradient([1])
+
+function _landscape!(mat, alg::DistanceGradient)
+    @assert maximum(alg.sources) <= length(mat)
+    @assert minimum(alg.sources) > 0
+    indices = vcat(CartesianIndices(mat)...)
```

Comment:
`reduce(vcat, CartesianIndices(mat))`  is a more performant version - the splatting will be very slow for matrices of any relevant size.